### ISSUE TITLE
Question RE integration with HA Power Dashboard

### ISSUE THREAD

**Issue by jwvanderbeck** [dotKrad/hass-fpl#30]
This isn't really a bug but it isn't feedback either so sorry wasn't sure how to classify it.  I did read the docs which got me all setup and running but I'm a little unclear on how best to integrate this into things like the energy dashboard in Home Assistant.

For a very simple system, just plain power draw from the grid (will be a while yest until Solar gets installed) what sensor do you feed into HA Energy?  I thought it would be the one named Meter Consumption `sensor.fpl_fpl8838681057deliveredreading` but that doesn't seem to be changing at all for me and I'm wondering if it only changes after a billing cycle is complete or something?  I saw another sensor named Billing Usage which SEEMS to be be updating maybe daily? Thought about using that instead but then realized it probably resets to 0 at the end of the bill so that would probably screw up the dashboard.

tl;dr what is the best way to hook this up to HA Energy? Which sensors should be used?

Thanks!

**Comment by jwvanderbeck** [comment]
From looking around it seems like maybe I should be using `sensor.fpl_dailydeliveredkwh` but that just shows Unknown for me :(

**Comment by gouthamravee** [comment]
I think there's something wrong with either the intergation or FPL. For a few weeks now I haven't received any new power related info from the integration, only some billing data. Everything else in the integration shows up as Unknown. 

I've fully uninstalled the addon through HACS and reinstalled with no change. 
I also checked logs for any errors but no signs there either.

**Comment by zoopster** [comment]
it's always best to open a new issue with these...my integration is working beautifully and unless you updated it with a newer version (there have been no updates in a while), it's likely a login or some other issue unrelated to the integration.

**Comment by dotKrad** [comment]
> I think there's something wrong with either the intergation or FPL. For a few weeks now I haven't received any new power related info from the integration, only some billing data. Everything else in the integration shows up as Unknown.
> 
> I've fully uninstalled the addon through HACS and reinstalled with no change. I also checked logs for any errors but no signs there either.

the same happened to me a few weeks ago, there was only a 9.01 value in the projected bill, all other sensors were unknown , it fixed itself next billing cycle

**Comment by gouthamravee** [comment]
@dotKrad 

I've gone through at least 1 billing cycle with the problem, unfortunately I have to wait till the 28th for the next billing cycle. Do you think this is an issue on the FPL side? 

I know their site was giving me problems this and last week, every time I logged in it would say page not found and give me options for the different regions. 

Anything you know of that I could do to get the integration to pull the other data fields again?

**Comment by dotKrad** [comment]
its quite hard to keep track of the changes, I usually change the integration when I get an error, FPL doesn't have a public API, they don't support 3rd parties,

**Comment by jwvanderbeck** [comment]
While all this is certainly interesting it doesn't address my topic of what sensors should be used to integrate with the HA Energy panel?  I'm still not understanding that after all this time.

**Comment by zoopster** [comment]
@jwvanderbeck While I don't use the FPL integration with the energy dashboard...you can configure it at http://homeassistant:8123/config/energy/dashboard. FPL doesn't provide good data for a total daily (they're only interested in monthly for billing) you can use the daily_usage sensor, but it's pretty unreliable. (it's a FPL issue, not an integration issue). I use the FPL integration in a card to keep tabs on what FPL thinks is my bill-to-day usage (similar to the energy dashboard on fpl.com and use sensors (split-ring current transformers to a circuitsetup card through ESPHome) on my electrical panel to provide energy dashboard actual usage.

**Comment by gouthamravee** [comment]
@jwvanderbeck I had to create a utility helper the FPL total KWh sensor. The one that is always increasing so its a total reading of your kwh usage. I undid all of my setup because of the recent issues I've had, once I get it setup again I will update with my exact setup. Like Zoopster said though, the daily readings are unreliable, if you want super reliable power readings you're better off setting up something in your electrical panel. 

@dotKrad figured out why my integration stopped working, my power meter is broken. Random contractor showed up yesterday to replace it, FPL didn't even give me a heads up. Had to call them to verify what was going on.

**Comment by CamiloValderruten** [comment]
Documented in the README **Energy Dashboard** section (v1.0.3+):

- Use external statistics `fpl:<account>_hourly_usage` and `fpl:<account>_hourly_cost` (add via the chart icon in Energy → Electricity grid consumption)
- Do **not** use Meter Consumption / Billing Usage sensors for the dashboard — those are billing-cycle or daily totals, not continuous grid consumption
- Hourly Usage **sensors** stay Unknown by design; real hourly data is in external statistics

https://github.com/dotKrad/hass-fpl#energy-dashboard

**Comment by CamiloValderruten** [comment]
Answered in README Energy Dashboard docs.
